queue_test.go

```diff
@@ -324,6 +329,56 @@ func TestQueue_Turbo(t *testing.T) {
 	}
 }
 
+func TestQueue_NewFlock(t *testing.T) {
```

Comment:
Can you add a test where you Close() the queue and then attempt to Enqueue() and Dequeue() on the same queue instance?  I'd expect an error to occur in each of those situations.

Reply:
Yes! Great proposal. Done.